From julia@diku.dk  Tue May 12 09:01:22 2009
From: Julia Lawall <julia@diku.dk>
Date: Tue, 5 May 2009 15:47:06 +0200 (CEST)
Subject: Staging: wlan-ng: p80211wext.c: problem with IW_ENCODE_ALG_WEP
To: Jiri Slaby <jirislaby@gmail.com>
Cc: richard@rsk.demon.co.uk, gregkh@suse.de
Message-ID: <Pine.LNX.4.64.0905051545450.9552@pc-004.diku.dk>


From: Julia Lawall <julia@diku.dk>

The expression !(ext->alg & IW_ENCODE_ALG_WEP) appears to be incorrect,
because there are several possible values for ext->alg that give 1 when
bit-anded with IW_ENCODE_ALG_WEP.  Therefore Richard Kennedy suggested to
rewrite the code with !=

Added \n at the end of the debug string as well.

Signed-off-by: Julia Lawall <julia@diku.dk>
Cc: Jiri Slaby <jirislaby@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/staging/wlan-ng/p80211wext.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/staging/wlan-ng/p80211wext.c
+++ b/drivers/staging/wlan-ng/p80211wext.c
@@ -1487,8 +1487,8 @@ static int p80211wext_set_encodeext(stru
 	}
 
 	if (ext->ext_flags & IW_ENCODE_EXT_SET_TX_KEY) {
-		if (!(ext->alg & IW_ENCODE_ALG_WEP)) {
-			pr_debug("asked to set a non wep key :(");
+		if (ext->alg != IW_ENCODE_ALG_WEP) {
+			pr_debug("asked to set a non wep key :(\n");
 			return -EINVAL;
 		}
 		if (idx) {
